dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderUtility.h

```diff
@@ -70,3 +73,55 @@ inline PCWSTR GetWinSoundEventString(AppNotificationBuilder::AppNotificationSoun
         return L"ms-winsoundevent:Notification.Default";
     }
 }
+
+inline std::wstring Encode(winrt::hstring const& value)
+{
+    std::wstring encodedValue{};
+    for (auto ch : value)
+    {
+        switch (ch) {
+        case '%':
+            encodedValue.append(c_encodedPercent);
+            break;
+        case ';':
+            encodedValue.append(c_encodedSemicolon);
+            break;
+        case '=':
+            encodedValue.append(c_encodedEquals);
+            break;
+        default:
+            encodedValue.push_back(ch);
+        }
+    }
+    return encodedValue;
+}
+
+// This function replaces the encoded values with the actual values
+inline std::wstring DecodeString(std::wstring value, std::wstring const& encodedValue, char ch)
```

Comment:
You could define a reverse map to decode. ie. translate 3B to =. By doing that, you would simplify this logic even further and you wouldn't have to pass ch here since the map does it for you.